### DISCUSSION TITLE
Some question about the possibility of the combination of grain_growth and  mechanical modules

### DISCUSSION THREAD

**Discussion by zengfy-hust** [idaholab/moose#18171] [Q&A General] [answered]
Hello Moose workers:
          I recent use the dream3d to as the  input file for grain growth ,the reasults seemed well . however ,I encountered some problem which may seem stupid ,hop you can help me with it :
         First ,I want to use python scrpit to deal with the output file of moose in the format of .csv  .to calculate the grain diameter of each grain ,I set the output options as followed :
[outputs.txt](https://github.com/idaholab/moose/files/6715229/outputs.txt)
however ,the grain volumes stays unchangeded in the outputs .could you help me with this ?
       second ,I wander if grain_growth can combined with mecahnical ,which would be better fit for my paper .was is possible now ?
ps . In grain_growth.i the euler angle was read but i wander what use was it for in the phase field calculation? I searched the introduction file of moose ,but nothing was found about it .can you specified the use of it  ?
thank you so much

**Comment by zengfy-hust** [comment, marked as answer]
[test.zip](https://github.com/idaholab/moose/files/6716157/test.zip)
FYI，I provide the csv file of my simulation ,but the grain volume did not change at all ,I just wonder why .

**Reply by GiudGiud** [reply]
@dschwen @laagesen

**Reply by amjokisaari** [reply]
Please provide individual files instead of a .zip file.  We (edit) should not open it for computer security reasons.  Please provide a full copy of your input file.  

Grain growth can be combined with mechanics.  The elastic energy driving force can be incorporated into the Allen-Cahn equation.  Depending on your exact specific needs, the implementation may already exist in the phase field module.  

In grain_growth.i, the Euler angles are read to provide each grain with a crystallographic rotation with respect to the reference frame of the simulation.  This allows the elastic stiffness tensor and any eigenstrains or other directional properties to be rotated along with the grain.

**Reply by dschwen** [reply]
We need to see your input as Andrea said (by the way I can open the zip just fine). Take a look at `FeatureVolumeVectorPostprocessor`. That can give you locations and volumes of grains.

**Reply by amjokisaari** [reply]
Risky click, @dschwen.  No way am I opening some random zip file on the internet!

**Reply by zengfy-hust** [reply]
Thank you for your help .I successfully obtained the grain volume . @dschwen ,by he way ,I want to know your timetable about fixing the module moose/modules/combined/examples/phase_field-mechanics/EBSD_reconstruction_grain_growth_mech.i ,which is exactly  the module I want to use .
